charts/safe-wallet-web/values.yaml

```diff
@@ -33,18 +33,15 @@ commonAnnotations: {}
 nodeSelector: {}
 
 ## @param resources Resource specification to deployment
+## The docker image compiles code at the entrypoint and requires more resources. An OOMKilled error will occur if resources are lower than 4Gi
 ##
-resources: {}
-    # We usually recommend not to specify default resources and to leave this as a conscious
-    # choice for the user. This also increases chances charts run on environments with little
-    # resources, such as Minikube. If you do want to specify resources, uncomment the following
-    # lines, adjust them as necessary, and remove the curly braces after 'resources:'.
-    # limits:
-    #   cpu: 100m
-    #   memory: 128Mi
-    # requests:
-    #   cpu: 100m
-    #   memory: 128Mi
+resources:
```

Comment:
Can we keep the request small but the limit high? The problem is the large memory block is only used on startup, but not after that.

Reply:
We can, but that will provoke more probability that the OOM will occur, as Kubernetes will be able to schedule the pod to more servers. And I will need to manually reschedule pods between servers so this pod can enter into a server with enough room. This same scenario happens on Forum where it compiles at the entrypoint. The solution would be to talk with Safe team and ask them to fix it.

Reply:
I've come up with a workaround that might work. I've attached a dummy initContainer which allocates the required resources for compilation, so we assure that the Kubernetes node have space for it. The initContainer finishes immediately and the the main container only requests the low resources during normal operation and not for compiling.